orderbook/orderlist.py

```diff
@@ -87,6 +87,10 @@ def move_to_tail(self, order):
 
         order.next_order.prev_order = order.prev_order
 
+        # Added to resolved issue #16
+        order.prev_order = self.tail_order
+        order.next_order = None
```

Comment:
Issue #16 is caused by not setting the `prev_order` & `next_order` variables of the current `order` to point to the new locations.